pkg/kp/pcstore/consul_store_test.go

```diff
@@ -18,6 +18,85 @@ import (
 
 func TestCreate(t *testing.T) {
 	store := consulStoreWithFakeKV()
+	GetGoodCreatePodCluster(store, t)
+}
+
+func TestUpdate(t *testing.T) {
+	store := consulStoreWithFakeKV()
+	oldPc := GetGoodCreatePodCluster(store, t)
+
+	// After creating the pod cluster, we now update it
+	newPodID := types.PodID("pod_id-diff")
+	newAz := fields.AvailabilityZone("us-west-diff")
+	newClusterName := fields.ClusterName("cluster_name_diff")
+
+	newSelector := klabels.Everything().
+		Add(fields.PodIDLabel, klabels.EqualsOperator, []string{newPodID.String()}).
+		Add(fields.AvailabilityZoneLabel, klabels.EqualsOperator, []string{newAz.String()}).
+		Add(fields.ClusterNameLabel, klabels.EqualsOperator, []string{newClusterName.String()})
+
+	newAnnotations := fields.Annotations(map[string]interface{}{
+		"bar": "foo",
+	})
+
+	intendedPc := fields.PodCluster{
+		ID:               oldPc.ID,
+		PodID:            newPodID,
+		AvailabilityZone: newAz,
+		Name:             newClusterName,
+		PodSelector:      newSelector,
+		Annotations:      newAnnotations,
+	}
+
+	newSession := kptest.NewSession()
+	newPc, err := store.Update(intendedPc, 0, newSession)
+
+	if err != nil {
+		t.Fatalf("Unable to update pod cluster: %s", err)
+	}
+
+	if newPc.ID == "" {
+		t.Errorf("pod cluster should have an id")
+	}
+
+	if newPc.PodID == "" {
+		t.Errorf("pod cluster should have a pod id")
+	}
+
+	if newPc.ID != oldPc.ID {
+		t.Errorf("id should not have been updated. Wanted '%s' got '%s'", oldPc.ID, newPc.ID)
+	}
+
+	if newPc.PodID != newPodID {
+		t.Errorf("pod id wasn't properly updated. Wanted '%s' got '%s'", newPodID, newPc.PodID)
+	}
+
+	if newPc.AvailabilityZone != newAz {
+		t.Errorf("availability zone wasn't properly updated. Wanted '%s' got '%s'", newAz, newPc.AvailabilityZone)
+	}
+
+	if newPc.Name != newClusterName {
+		t.Errorf("cluster name wasn't properly updated. Wanted '%s' got '%s'", newClusterName, newPc.Name)
+	}
+
+	newTestLabels := klabels.Set{
+		fields.PodIDLabel:            newPodID.String(),
+		fields.AvailabilityZoneLabel: newAz.String(),
+		fields.ClusterNameLabel:      newClusterName.String(),
+	}
+
+	if matches := newPc.PodSelector.Matches(newTestLabels); !matches {
+		t.Errorf("the pod cluster has a bad pod selector")
+	}
+
+	if newPc.Annotations["bar"] != "foo" {
+		t.Errorf("Annotations didn't match expected")
+	} else if _, ok := newPc.Annotations["foo"]; ok {
+		t.Errorf("Annotations didn't match expected")
+	}
+}
+
+func GetGoodCreatePodCluster(store *consulStore, t *testing.T) fields.PodCluster {
```

Comment:
I think this makes more sense as a private method called `createPodCluster`. No need to denote it as "good" or say that we "get" it as a return type. Those facts are implied.